### PR TITLE
FIX-#4422: get rid of case sensitivity for warns_that_defaulting_to_pandas

### LINKED ISSUE #4422: `warns_that_defaulting_to_pandas` is case sensetive
`warns_that_defaulting_to_pandas` utility function that is used to wrap defaulting-to-pandas pieces of code in tests appears to be a case sensetive:
https://github.com/modin-project/modin/blob/0ab04e104a60419c464fc1833a13f7f2b92e7983/modin/test/test_utils.py#L254
The function only considers as a proper warning message the ones that starts with a small "d", however there're also cases when the message starts with a capital "D":
https://github.com/modin-project/modin/blob/0ab04e104a60419c464fc1833a13f7f2b92e7983/modin/error_message.py#L43-L46

### PR DESCRIPTION
**Pull request by dchigarev**
Signed-off-by: proxodilka <zeeron209@gmail.com>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #4422 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date 
- [x] added (Issue Number: PR title (PR Number)) and github username to release notes for next major release

### CODE REVIEW

File: docs/release_notes/release_notes-0.15.0.rst [hunk 1]
```diff
@@ -33,6 +33,7 @@ Key Features and Updates
   * FEAT-#4359: Add __dataframe__ method to the protocol dataframe (#4360)
 * Update testing suite
   * TEST-#4363: Use Ray from pypi in CI (#4364)
+  * FIX-#4422: get rid of case sensetivity for `warns_that_defaulting_to_pandas` (#4423)
```
Comment (mvashishtha) [inline docs/release_notes/release_notes-0.15.0.rst:36]: typo: sensitivity

**Review by mvashishtha — COMMENTED** [review]
Thank you for fixing this.

File: docs/release_notes/release_notes-0.15.0.rst (hunk 1 above)
Comment (dchigarev) [inline docs/release_notes/release_notes-0.15.0.rst:36]:
```suggestion
  * FIX-#4422: get rid of case sensitivity for `warns_that_defaulting_to_pandas` (#4423)
```

**Review by mvashishtha — APPROVED** [review]

**Review by YarShev — APPROVED** [review]
@dchigarev, LGTM, thanks!

**Pull request merged**